### ISSUE TITLE
Adding urls/pairs to the context without RoCrateBuilder

### ISSUE THREAD

**Issue by Code42Cate** [kit-data-manager/ro-crate-java#36]
There is no way to add to the the context without using the RoCrateBuilder (which is not possible when using the RoCrateReader).

There are probably more features missing, and the nicest fix IMO would be to extend the RoCrateBuilder to allow using an existing RoCrate as a template.
```java
RoCrate crate = new RoCrate.RoCrateBuilder(existingCrate)[...]build();
```

**Comment by Code42Cate** [comment]
I have not tested everything, but it seems like this constructor would be enough to enable my suggestion:
```java
public RoCrateBuilder(RoCrate crate) {
      this.payload = crate.roCratePayload;
      this.metadataContext = crate.metadataContext;
      this.preview = crate.roCratePreview;
      this.rootDataEntity = crate.rootDataEntity;
      this.jsonDescriptor = crate.jsonDescriptor;
      this.untrackedFiles = crate.untrackedFiles;
}
```

**Comment by Pfeil** [comment]
Seems like a good solution to me. If you do this, we could in theory remove all other setters of RoCrate, so modification is only possible from the builder, where we have potentially a better control over what is allowed and what is not. Although removing methods would be a breaking change, so we could mark it as deprecated and remove it in version 2.0, then. Thoughts?

**Comment by Code42Cate** [comment]
If we are sure that we don't lose any functionality, yes I would do that. Less interfaces that do the same:)

**Comment by Code42Cate** [comment]
We also need to keep in mind that removing stuff only works on the RoCrate objects, not with the builder.

**Comment by Code42Cate** [comment]
The issue is fixed with #39  ! :)
